Gemfile

```diff
@@ -1,8 +1,14 @@
 source 'https://rubygems.org'
 
-gem "github-pages", '193', group: :jekyll_plugins
+gem 'html-proofer'
+gem 'jekyll', '~> 4.1.1'
 
-# enable tzinfo-data for local build
-# gem 'tzinfo-data', platforms: [:mingw, :mswin, :x64_mingw]
-gem 'jekyll-paginate'
-gem 'faraday', '0.15.4'
+# If you want to use GitHub Pages, remove the "gem "jekyll"" above and
+# uncomment the line below. To upgrade, run `bundle update github-pages`.
+# gem "github-pages", group: :jekyll_plugins
```

Comment:
Any reason to not use github-pages and add jekyll above? it we are deploying in GitHub pages environment doesn't it make sense to use it?

Reply:
> Any reason to not use github-pages 

I think we are aware now that `github-pages` bring a lot of dependent gems. However, what we require is just `jekyll` and our required plugins.  See https://github.com/github/pages-gem/blob/master/lib/github-pages/dependencies.rb

> it we are deploying in GitHub pages environment doesn't it make sense to use it?

I think it has no connection with the environment. `github-pages` is a gem with a collection of other gems which GitHub thought of giving users as a package to make thing simple and easy.